docs/Q-and-A.md

```diff
@@ -0,0 +1,649 @@
+_Could we contribute to the repository?_
+
+it is in the Decentralized Identity Foundation's GitHub depository, [Sidetree](https://github.com/decentralized-identity/sidetree) is the name of it. So you can contribute code there, you can read every line of code. There's no code insertions after the fact, like every bit of every piece of code that we run, as an ION node, is there. So it's totally inspectable.
+- [Contribution guidelines for the project](https://github.com/decentralized-identity/sidetree/blob/master/docs/contribution-guidlines.md).
+- [The Sidetree release process](https://github.com/decentralized-identity/sidetree/blob/master/docs/release-process.md)
+- [The Sidetree coding style guide](https://github.com/decentralized-identity/sidetree/blob/master/docs/styleguide.md)
+
+# Q&A section Sidetree operational
+
+## *Q:  Where can I download Sidetree?
+On (sub)page(s of) [github](https://github.com/decentralized-identity/sidetree)
+
+##  *Q: Where can we find the code and how could a coder get started?
+The homepage on github [README.md](../README.md) pretty much sums up all the possibilities to download the available code and how developers can engage in the development process currently. We welcome contributions.
+
+##  **Q: What are the external dependencies of Sidetree?
+_I hear ya: "Sidetree is able to do all this **without requiring** trusted intermediaries, centralized authorities, special protocol tokens, or secondary consensus mechanisms, while preserving the core attributes of decentralization and immutability of the underlying ledger systems it is implemented on." But what does Sidetree require instead?_
+
+Sidetree is dependent of:
+1. A content addressable storage layer (e.g., IPFS)
+2. Reasonable fees and confirmation-times on layer 1
+3. Consensus, transaction validation and governance in general of the layer 1 blockchain it settles on.
+4. Ongoing development and blockchain maintenance for safety of the records now and in the future.
+5. {TBW prio 2}
+_(@henkvancann)_
+
+##  **Q: In what can we find the efficiency booster that Sidetree promises to be?
+Sidetree nodes that participate in writing operations into the overlay network do so by **anchoring** Content-Addressable Storage (`CAS`) (e.g. IPFS) references to **aggregated** bundles of operations in an underlying ledger.
+
+##  ***Q: "Process operations in accordance with the protocol deterministic ruleset." What does this mean?
+Read this [section](https://identity.foundation/sidetree/spec/#late-publishing) that illustrates a DID owner, Alice, creating forks by creating and anchoring operations in the past that she does not expose to the network. Known as _Late Publishing_ of a DID operation.\
+This type of processing guarantees only one `fork` of a DID’s state history can ever be valid.
+
+##  *Q: What would you see as the main drawback of Sidetree?
+There aren't many. Sidetree has Self-Certifying Identifiers, however they are non-transferable. {Right?}
```

Comment:
Suggestions:

- For any given Sidetree based DID method, a user has to trust a Sidetree node operation, or run a Sidetree node themselves
- Batching introduces a throughput / latency tradeoff: the more infrequent batches are sent to the underlying ledger, the more latency there is compared to a direct ledger transaction
- Using multiple CAS files improves scalability but can slow down read operation depending on how the observer is implemented